modin/engines/dask/pandas_on_dask/frame/partition.py

```diff
@@ -114,24 +90,34 @@ def apply(self, func, **kwargs):
         -----
         The keyword arguments are sent as a dictionary.
         """
-        func = pkl.dumps(func)
-        call_queue = self.call_queue + [[func, kwargs]]
-        future = default_client().submit(
-            apply_list_of_funcs, call_queue, self.future, pure=False
-        )
-        futures = [default_client().submit(lambda l: l[i], future) for i in range(2)]
+        client = default_client()
+        call_queue = self.call_queue + [[func, args, kwargs]]
+        if len(call_queue) > 1:
+            future = client.submit(
+                apply_list_of_funcs, self.future, call_queue, pure=False
+            )
+        else:
+            # We handle `len(call_queue) == 1` in a different way because
+            # this improves performance a bit.
+            func, args, kwargs = call_queue[0]
+            future = client.submit(apply_func, self.future, func, *args, **kwargs)
+        futures = [
+            client.submit(lambda l, i: l[i], future, i, pure=False) for i in range(2)
+        ]
```

Comment:
I wonder if there's more elegant way of unpacking a "future tuple"...

Reply:
So far I have not seen it.